client/src/services/api.ts

```diff
@@ -120,14 +120,20 @@ function oidcFactory() {
     };
 }
 
+let isApiRequestInProgress = false;
+
 function metrics(url: string, cb: DataCallback<Metrics[]>) {
     const handel = setInterval(getMetrics, 10000);
     getMetrics();
 
     async function getMetrics() {
         try {
-            const metric = await request(url);
-            cb(metric.items || metric);
+            if (!isApiRequestInProgress) {
+                isApiRequestInProgress = true;
+                const metric = await request(url);
+                isApiRequestInProgress = false;
```

Comment:
I think we should add a `finally` block and put this in there. Otherwise if there's ever an error, `isApiRequestInProgress` will never get cleared and we would just stop loading metrics.

Reply:
Yes, good catch.